app/models/business.rb

```diff
@@ -17,6 +18,8 @@ class Business < ApplicationRecord
 
   alias_attribute :name, :contact_name
 
+  delegate :email, to: :referring_user, prefix: true, allow_nil: true
```

Comment:
Ohh, very nice! But let's abstract these two methods to a concern that can be included in `Developer` and `Business`.

Reply:
In order to do that the `belongs_to :user` association would have to be moved to the new concern. Are you OK with that?

Reply:
Why is that?

Reply:
I got the following error when I tried creating a concern without that association:

"Cannot have a has_many :through association 'Business#referring_user' which goes through 'Business#user' before the through association is defined."

The only options I can think of would be to move the "include" statements after the association is defined in the models, or to move the assocation into the new concern.

Reply:
Ah, gotcha. OK, I don't think it makes sense to pull that association into a concern because it is core to the developer/business model.

One more idea, though. Can we define the association _twice_? Once in the model and again in the concern? If so maybe that could work. If not, then lets skip the refactor and roll with what you originally proposed.

Reply:
I tried defining the association in both the model and the concern, and the error kept happening.

Reply:
Gotcha. Let’s punt on the reactor then. Thanks for diving in!

Reply:
I reverted the changes